server/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -236,20 +235,20 @@ public SnapshotInfo snapshot(final String repositoryName, final SnapshotId snaps
      * @param request  snapshot request
      * @param listener snapshot creation listener
      */
-    public void createSnapshot(final SnapshotRequest request, final CreateSnapshotListener listener) {
-        final String repositoryName = request.repositoryName;
-        final String snapshotName = request.snapshotName;
+    public void createSnapshot(final CreateSnapshotRequest request, final CreateSnapshotListener listener) {
+        final String repositoryName = request.repository();
+        final String snapshotName = indexNameExpressionResolver.resolveDateMathExpression(request.snapshot());
```

Comment:
we resolve this twice now, and if you have stuff like `now` in it, it might resolve to two different values. Perhaps we can wrap the original request?

Reply:
How about this instead https://github.com/elastic/elasticsearch/pull/37464/commits/27ca06bf97b6477bd3da3dc92db7d30be3d08a6f ? Just pass the snapshot back in the callback and get a cleaner `equals` in the transport action as well? :)

Reply:
++